TogglTrack/Core/Reducers/LoginReducer.swift

```
@@ -103,10 +97,17 @@ private func subscribePushNotificationTokenToTogglEffect(_ api: APIProtocol, _ t
         .eraseToEffect()
 }
 
-private func subscribePushNotificationTokenToFCMEffect(_ firebaseAPI: FirebaseAPIProtocol, _ apnToken: String) -> Effect<AppAction>
+private func subscribePushNotificationTokenToFCMEffect(_ api: APIProtocol, _ firebaseAPI: FirebaseAPIProtocol, _ apnToken: String) -> Effect<AppAction>
 {
     firebaseAPI.getFCMToken(for: FCMPushToken(apnToken))
-        .map { response in .user(.subscribedPushNotificationTokenToFCM(response)) }
+        .tryFlatMap { (response) -> AnyPublisher<AppAction, Error> in
+            guard let fcmToken = response.fcmTokens?.first else { return Fail(error: UserError.NoFCMResponse).eraseToAnyPublisher() }
+            guard fcmToken.status == "OK" else { return Fail(error: UserError.FailedToGetFCMToken(fcmToken.status)).eraseToAnyPublisher() }
+            guard let togglToken = fcmToken.toTogglPushToken() else { return Fail(error: UserError.NoFCMTokenAvailable).eraseToAnyPublisher() }
```

On these three you can now just throw instead of returning a Fail Publisher, thanks to `tryFlatMap`